Python/ceval.c

```diff
@@ -3471,25 +3471,28 @@ _PyEval_ImportNameWithImport(PyThreadState *tstate, PyObject *import_func, PyObj
                         ilevel);
     }
 
-    PyObject* args[5] = {name, globals, locals, fromlist, level};
+    PyObject *args[5] = {name, globals, locals, fromlist, level};
     PyObject *res = PyObject_Vectorcall(import_func, args, 5, NULL);
     return res;
 }
 
 static int
-check_lazy_import_comatibility(PyThreadState *tstate, PyObject *globals,
+check_lazy_import_compatibility(PyThreadState *tstate, PyObject *globals,
                                PyObject *name, PyObject *level)
 {
-     // Check if this module should be imported lazily due to the compatbility mode support via
-    // __lazy_modules__.
+     // Check if this module should be imported lazily due to
+     // the compatibility mode support via __lazy_modules__.
     PyObject *lazy_modules = NULL;
     PyObject *abs_name = NULL;
     int res = -1;
 
     if (globals != NULL &&
-        PyMapping_GetOptionalItem(globals, &_Py_ID(__lazy_modules__), &lazy_modules) < 0) {
+        PyMapping_GetOptionalItem(globals, &_Py_ID(__lazy_modules__), &lazy_modules) < 0)
+    {
         return -1;
-    } else if (lazy_modules == NULL) {
+    }
+    if (lazy_modules == NULL) {
+        assert(!PyErr_Occurred());
```

Comment:
This split is also not really necessary but I think it helps for readability purposes (if globals is NULL, then we move to lazy_modules check and return; if globals is not NULL, and if `GetOptionalItem` is -1, we return and we're done; finally, if `lazy_modules` is set (either to NULL or something else) the first if condition would fail and then we would move to the `lazy_modules == NULL` check).